src/interface/tasks/tasks.controller.ts

```diff
@@ -1,14 +1,42 @@
 import { Controller, Logger } from '@nestjs/common'
 import { Ctx, MessagePattern, NatsContext, Payload } from '@nestjs/microservices'
+import { GraphQLRequest } from 'apollo-server-core'
 
+import { GenericActivity } from '@adapters/activity/activities/generic-activity'
+import { UserInterface } from '@core/modules/user/user.interface'
 import { isOfTypeCommand } from '@core/ports/commands/command.factory'
 import { CorePortsProvider } from '@core/ports/ports.provider'
+import { NotificationProvider } from '@infrastructure/notification/notification.provider'
 
 @Controller()
 export class TasksController {
   private readonly logger = new Logger(TasksController.name)
 
-  constructor(private readonly corePorts: CorePortsProvider) {}
+  constructor(
+    private readonly corePorts: CorePortsProvider,
+    private readonly notification: NotificationProvider,
+  ) {}
+
+  @MessagePattern('notification-ports.*')
+  async useNotification(@Payload() data: any, @Ctx() natsContext: NatsContext) {
+    const messageName = natsContext.getSubject()
+    const [_, commandName] = messageName.split('.')
+
+    const user = await this.corePorts.dispatchCommand<UserInterface>('get-user', {
+      id: data.userId,
+    })
+    const context = {
+      userWithContext: user,
+    }
+    const a = new GenericActivity<unknown, unknown>(
+      commandName,
+      data,
+      context as Partial<GraphQLRequest>,
+      undefined,
+    )
+
+    await this.notification.dispatch(a)
```

Comment:
Mudança opcional (caso negativo, transformar em débito técnico):

Mudar o nome da variável `a`.

Minha sugestão:
```suggestion
    const activity = new GenericActivity<unknown, unknown>(
      commandName,
      data,
      context as Partial<GraphQLRequest>,
      undefined,
    )

    await this.notification.dispatch(activity)
```